src/test/e2e/api/proxy/proxy.e2e.test.ts

```diff
@@ -991,3 +992,47 @@ test('should return maxAge header on options call', async () => {
             expect(res.headers['access-control-max-age']).toBe('86400');
         });
 });
+
+test('should terminate data polling when stop is called', async () => {
+    const frontendToken = await createApiToken(ApiTokenType.FRONTEND);
+    const user = await app.services.apiTokenService.getUserForToken(
+        frontendToken.secret,
+    );
+
+    const logTrap = [];
+    const getDebugLogger = (): Logger => {
+        return {
+            /* eslint-disable-next-line */
+            debug: (message: any, ...args: any[]) => {
+                logTrap.push(message);
+            },
+            /* eslint-disable-next-line */
+            info: (...args: any[]) => {},
+            /* eslint-disable-next-line */
+            warn: (...args: any[]) => {},
+            /* eslint-disable-next-line */
+            error: (...args: any[]) => {},
+            /* eslint-disable-next-line */
+            fatal: (...args: any[]) => {},
+        };
+    };
+
+    /* eslint-disable-next-line */
+    const proxyRepository = new ProxyRepository(
+        {
+            getLogger: getDebugLogger,
+            frontendApi: { refreshIntervalInMs: 1 },
+        },
+        db.stores,
+        app.services,
+        user,
+    );
+
+    await proxyRepository.start();
+    proxyRepository.stop();
+    // Polling here is an async recursive call, so we gotta give it a bit of time
+    await new Promise((r) => setTimeout(r, 10));
```

Comment:
Had a very good suggestion to tackle this test with fake timers. However, that never seems to surface the bug on main. I believe that's because this is fundamentally a race condition. I think this is okay since it only adds a 10ms wait here but I'm open to suggestions